### ISSUE TITLE
Basic Print Function

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/ts-android#883]
Users should be able to print their translation from tS.  Likely, we'll just want to generate a simple PDF that can then be sent to another app for printing.

**Comment by neutrinog** [comment]
+ for obs allow them to print without images and print with images. Internet will be needed when printing with images so we can download them.
+ allow printing just chunks that are done, or print everything.
+ so we will have two checkboxes, include images, and include just done.

**Comment by neutrinog** [comment]
where do you want this print functionality to be available? At the end of the publish process, in backup, in it's own menu  item?

**Comment by neutrinog** [comment]
@jag3773 do we have any specs on how the pdf should be rendered? e.g. placement of chapter titles, how do we title chapters (`Chapter #` maybe), margin, will frames each be their own paragraph or joined together? What delimiter should we use for joining frames? etc.

Note to self: pdf library tutorial at http://www.vogella.com/tutorials/JavaPDF/article.html

**Comment by jag3773** [comment]
For OBS, we have PDFs here: https://unfoldingword.org/stories/ that can serve as a guide.

For Scripture, we are working on PDF generation right now, so we don't have anything concrete.  I could pass you some dated examples if that would help?

Print should probably be a separate entry in the menu when you have a project open.  Or, we could put it in the "i" pop up from the main screen for each project?  @sampeckham what do you think?

**Comment by neutrinog** [comment]
@jag3773 do you want to print only chunks marked as done?

**Comment by neutrinog** [comment]
@jag3773 what margins do you want on the top and bottom of pages?

**Comment by sampeckham** [comment]
We discussed this being in the 'i' modal screen, from the Home screen. That makes more sense and give the context of printing the whole thing (rather than just a chunk or chapter).

Here's how that modal should then look. @neutrinog Do you want a new ticket for the modal design?
![screen shot 2015-11-16 at 14 03 24](https://cloud.githubusercontent.com/assets/323212/11183695/44d76e6e-8c6b-11e5-9f14-738ceb5e5f32.png)

**Comment by sampeckham** [comment]
Added an item for the modal here: #1010

**Comment by jag3773** [comment]
@neutrinog @sampeckham Let's have a toggle for printing all chunks or only chunks marked as done.  Default to chunks marked as done.  Something like:

* [x] Only print chunks marked as "Done"

**Comment by jag3773** [comment]
@neutrinog Let's use 1 inch margins.

**Comment by neutrinog** [comment]
@jag3773 what image resolution do you want to use?

**Comment by neutrinog** [comment]
I'm going to assume 360px. That's only 36M zipped as apposed to the larger size which is 770M

**Comment by neutrinog** [comment]
@jag3773 also, we have two options for retrieving the images. We could download the entire zipball (36M) in one shot which would be faster and more efficient, or download one at a time. If we do one at a time we could more easily recover from an interrupted internet connection, or just download images for the frames that will be printed. What's your preference?

**Comment by neutrinog** [comment]
this is related to https://github.com/unfoldingWord-dev/ts-android/issues/886. We'll access the same images.

**Comment by neutrinog** [comment]
@jshuma downloading should occur in `core.Downloader` and to initiate a download the user will have checked "include images" when printing a translation. Later we'll add the ability to download images while viewing a translation, see https://github.com/unfoldingWord-dev/ts-android/issues/886. In order to keep the ui from freezing the new download method should be called in a new task which you should create in `com.door43.translationstudio.tasks`. I can walk you through how to use the task manager.

we'll need to display a dialog with a progress bar. after the download is done the printing will begin.

**Comment by jshuma** [comment]
Added a [pull request](https://github.com/unfoldingWord-dev/ts-android/pull/1234) for this one. It includes an Internet connection usage warning, a progress dialog, downloading, extraction, and avoiding re-downloading images that were already downloaded.

I was able to verify that this all works (except that I didn't get to verify the "download failed" use case), and the images are in sync with the latest zipfile reality, but I have two concerns:
* I'm a little uneasy about the inclusion of a `360px` directory under `IMAGES_DIR` (see 2de9a865bbf34d475a256de2fbfe0cf6e858a4cd... it's there because the zipfile includes this directory). An alternative fix would be to modify the zipfile to keep everything at the top level, but I'm not sure whether this might break any existing users of the zipfile.
* `PdfPrinter.addContent()` is able to load these files and attempt to insert them in the PDF. However, I didn't spot any images in the output PDF. It's getting as far as `PdfPrinter.addImage()` getting a success result from `Document.add()` so I wonder if there might be something wrong with image inclusion being set up. It really seems to be at least getting the file.
